### PR TITLE
Upgrade to Rails 5.0.0 and ruby 2.3.1

### PR DESCRIPTION
**Pull request by delkopiso**
- Remove `spring`, `jazz_hands`, `pry-byebug` gem
- Remove `rails-api` gem since it's now bundled with `rails` itself
- Remove `Guardfile` since guard wasn't installed anyway
- Fix deprecations warnings
- Remove and ignore `.ruby-version`
- Remove `rubocop.yml`
- Use `sinatra` master, and include `sinatra/rack-protection` as a result
- Add ci environment file
- Turn off static asset server in test environment

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,22 +15,19 @@ gem 'honeybadger'
 gem 'skylight'
 gem 'rack-cors'
 gem 'sidekiq'
-gem 'sinatra', require: nil
+gem 'sinatra', github: 'sinatra', require: nil
```
Comment (abepetrillo) [inline Gemfile]: Never seen the `github` reference before, is that new?

File: Gemfile [hunk 2]
```diff
@@ -16,22 +15,19 @@ gem 'honeybadger'
...
 group :development do
-  gem 'spring'
   gem 'bullet'
 end
 
 group :test, :development do
-  gem 'jazz_hands', github: 'nixme/jazz_hands', branch: 'bring-your-own-debugger'
-  gem 'pry-byebug'
+  gem 'pry'
+  gem 'pry-rails'
```
Comment (abepetrillo) [inline Gemfile]: Actually pry-byebug is better, you can't use `next` etc with `pry-rails`

File: config/environments/test.rb [hunk 3]
```diff
@@ -13,8 +13,8 @@
   config.eager_load = false
 
   # Configure static asset server for tests with Cache-Control for performance.
-  config.serve_static_files  = true
-  config.static_cache_control = 'public, max-age=3600'
+  config.public_file_server.enabled = false
+  # config.public_file_server.headers = { 'Cache-Control' => 'public, max-age=3600' }
```
Comment (abepetrillo) [inline config/environments/test.rb]: Just delete this?

**Comment by abepetrillo** [comment]
:+1: Minor comments

File: Gemfile (hunk 1 above)
Comment (delkopiso) [inline Gemfile]: Not sure how new it is. But it's in the docs: http://bundler.io/man/gemfile.5.html#GITHUB-github-

File: Gemfile (hunk 2 above)
Comment (delkopiso) [inline Gemfile]: I didn't know that

**Pull request merged**